addon/serializers/json-api.js

```diff
@@ -202,6 +202,18 @@ const JSONAPISerializer = JSONSerializer.extend({
     return normalizedPayload;
   },
 
+  normalizeQueryRecordResponse() {
+    let normalized = this._super(...arguments);
+
+    if (Array.isArray(normalized.data)) {
+      assert('Expected the primary data returned by the serializer for a `queryRecord` response to be a single object but instead it was an array.', false, {
+        id: 'ds.serializer.json-api.queryRecord-array-response'
+      });
```

Comment:
Instead of wrapping this in an if block should we replace `false` with `!Array.isArray(normalized.data)` so it gets stripped correctly is a prod build?